course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt

```diff
@@ -241,6 +242,34 @@ private fun CourseSectionScreen(
                             }
                         }
 
+                        is CourseSectionUIState.Gated -> {
+                            Column(
+                                modifier = Modifier.fillMaxSize(),
+                                verticalArrangement = Arrangement.Center,
+                                horizontalAlignment = Alignment.CenterHorizontally
+                            ) {
+                                Image(
+                                    painter = painterResource(id = R.drawable.ic_course_gated),
+                                    contentDescription = "gated",
+                                    modifier = Modifier.size(48.dp)
+                                )
+                                Spacer(modifier = Modifier.height(16.dp))
+                                Text(
+                                    modifier = Modifier
+                                        .padding(horizontal = 16.dp)
+                                        .fillMaxWidth(),
+                                    text = stringResource(
+                                        id = R.string.course_gated_subsection,
+                                        uiState.prereqSubsectionName ?: ""
+                                    ),
+                                    textAlign = TextAlign.Center,
+                                    style = MaterialTheme.appTypography.titleMedium,
+                                    color = MaterialTheme.appColors.textPrimary,
+
```

Comment:
Could we add a button to navigate to the correct subsection similar to the web UI. For the button we could implement an onClick function similar to [this one](https://github.com/open-craft/openedx-app-android/blob/b542698f99cd29ca79d5de1bbb5641c9b954ee4f/course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineScreen.kt#L92-L103) ?